.github/workflows/generate-preview-link.yml

```diff
@@ -0,0 +1,134 @@
+name: Generate preview link
+
+permissions:
+    actions: write
+    checks: write
+    contents: write
+    deployments: write
+    pull-requests: write
+    statuses: write
+
+on:
+    workflow_run:
+        workflows: ['Pre-generate preview link']
+        types:
+            - completed
+
+env:
+    NODE_OPTIONS: '--max-old-space-size=8192'
+
+concurrency:
+    group: cloudflare-pages-build-${{ github.event.workflow_run.head_branch }}
+    cancel-in-progress: true
+
+jobs:
+    build_to_cloudflare_pages:
+        runs-on: Ubuntu-latest
+        if: >
+            github.event.workflow_run.event == 'pull_request' &&
+            github.event.workflow_run.conclusion == 'success'
+        steps:
+            - name: Download artifact
+              id: download-artifact
+              uses: dawidd6/action-download-artifact@v2
+              with:
+                  workflow_conclusion: success
+                  run_id: ${{ github.event.workflow_run.id }}
+                  name: 'pr-${{ github.event.workflow_run.id }}'
+
+            - name: Retrieve and verify user organization
+              id: pr_information
```

Comment:
This checks if the user is from binary-com organization, thus ensuring only binary-com user's PR can perform the build and deploy steps

Reply:
Worst case scenario, all PRs outside binary-com can be blocked and approved for workflow run first:
<img width="880" alt="Screenshot 2023-09-22 at 4 02 17 PM" src="https://github.com/binary-com/deriv-com/assets/103016120/a20cb140-ecaf-4fd4-bb96-af07de81c78a">